pkg/modelzooserver/modelzooserver.go

```diff
@@ -109,7 +109,60 @@ func (s *modelZooServer) ListModelDefs(ctx context.Context, req *pb.ListModelReq
 }
 
 func (s *modelZooServer) ListTrainedModels(ctx context.Context, req *pb.ListModelRequest) (*pb.ListTrainedModelResponse, error) {
-	return &pb.ListTrainedModelResponse{}, nil
+	var sql string
+	if req.Size <= 0 {
+		sql = fmt.Sprintf(`SELECT a.name, a.version, a.url, a.description, a.metrics, c.name, c.version FROM %s AS a
+LEFT JOIN %s AS b ON a.model_def_id=b.id
+LEFT JOIN %s AS c ON b.model_coll_id=c.id;`,
+			trainedModelTable, modelDefTable, modelCollTable)
+	} else {
+		sql = fmt.Sprintf(`SELECT a.name, a.version, a.url, a.description, a.metrics, c.name, c.version FROM %s AS a
+LEFT JOIN %s AS b ON a.model_def_id=b.id
+LEFT JOIN %s AS c ON b.model_coll_id=c.id LIMIT %d OFFSET %d;`,
+			trainedModelTable, modelDefTable, modelCollTable, req.Size, req.Start)
+	}
+	rows, err := s.DB.Query(sql)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	names := []string{}
+	modelVersions := []string{}
+	urls := []string{}
+	descs := []string{}
+	metrics := []string{}
+	imageurls := []string{}
+
+	for rows.Next() {
+		n := ""
+		v := ""
+		url := ""
+		desc := ""
+		m := ""
+		imagename := ""
+		imagetag := ""
+		if err := rows.Scan(&n, &v, &url, &desc, &m, &imagename, &imagetag); err != nil {
+			return nil, err
+		}
+		names = append(names, n)
+		modelVersions = append(modelVersions, v)
+		urls = append(urls, url)
+		descs = append(descs, desc)
+		metrics = append(metrics, m)
+		imageURL := fmt.Sprintf("%s:%s", imagename, imagetag)
+		imageurls = append(imageurls, imageURL)
+	}
+
+	return &pb.ListTrainedModelResponse{
+		Names:          names,
```

Comment:
can we return list of wrapped object install of lists of fields?

Reply:
Good idea!